cg/src/domains/management/subjects/subjects.controller.js

```diff
@@ -17,14 +17,15 @@ class SubjectsController {
   }
 
   async listSubjects(req, res) {
-    const subjectsPage = await this.subjectsService.listSubjects(req.query.page);
-    subjectsPage.map(subject => {
+    const subjectsInfo = await this.subjectsService.listSubjects(req.query.page);
+    subjectsInfo.subjects.map(subject => {
       // Covers the subjects data with '*'s
       _.mapObject(subject, (value, key) => {
         subject[key] = coverString(value);
+        return subject[key];
```

Comment:
basically the reason i raised the issue is that inputs inside map functions is a bit of an anti-pattern. The idea of map is you build a new object for every input value. Especially in javascript mutation of inputs can be really scary since everything is pass-by reference.

You will see the way map is used on the MDN docs is in a non-mutatey way, thats what I would like to see here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map